### PR TITLE
Split migrator job to sync and async

### PR DESCRIPTION
**Pull request by srikanthccv**
Fixes https://github.com/SigNoz/signoz/issues/5433

## Summary by CodeRabbit

- **New Features**
	- Introduced a new asynchronous schema migrator with updated configuration options.
	- Added service account, role, and role binding configurations for enhanced RBAC management.
	- Updated OpenTelemetry Collector and metrics configurations with new roles and bindings.
	- Enhanced the schema migrator's initialization process with a standardized command.

- **Bug Fixes**
	- Simplified initialization commands for schema migrator to standardize deployment processes.

- **Documentation**
	- Enhanced comments and annotations in the configuration files for better clarity and guidance.

- **Chores**
	- Updated image tags for schema migrator, OpenTelemetry Collector, and metrics to the latest versions.

### CODE REVIEW

File: charts/signoz/values.yaml [hunk 1]
```diff
@@ -1270,10 +1267,9 @@ schemaMigrator:
     tag: 0.102.10
     pullPolicy: IfNotPresent
 
-  args: {}
+  args:
+    - "--up="
```
Comment (prashant-shahi) [inline charts/signoz/values.yaml:1271]: is this valid flag?

Comment (srikanthccv) [reply charts/signoz/values.yaml:1271]: Yes

File: charts/signoz/values.yaml [hunk 2]
```diff
@@ -1267,13 +1264,12 @@ schemaMigrator:
...
-    tag: 0.102.10
+    tag: 0.111.2
     pullPolicy: IfNotPresent
 
-  args: {}
+  args:
+    - "--up="
   annotations:
-    "helm.sh/hook-weight": "1"
-    "helm.sh/hook-delete-policy": "before-hook-creation"
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply charts/signoz/values.yaml:1276]: https://github.com/SigNoz/charts/pull/526/files#diff-016ca27e3e4c84106ccd6f84fbb56b17fac0f8c9198bc8c9c4b5ac247b3cc28bR13

https://github.com/SigNoz/charts/pull/526/files#diff-a049b5acd82ec4d48db9ade4b5f448630167193f53e95d02bde597b823e9dc1bR13

[the remark this answers is not in the document]
Comment (prashant-shahi) [reply charts/signoz/values.yaml:1276]: In the past, I had tested `ttlSecondsAfterFinished` for the schema migrator. It didn't help much then.

**Review by prashant-shahi — APPROVED** [review]

**Pull request merged**